stdlib/public/runtime/ImageInspectionCOFF.cpp

```diff
@@ -14,37 +14,132 @@
 
 #include "ImageInspection.h"
 #include "ImageInspectionCOFF.h"
+#include "swift/Runtime/Metadata.h"
 
 #if defined(__CYGWIN__)
 #include <dlfcn.h>
 #endif
+#include <memory>
 
 using namespace swift;
 
 namespace {
-static const swift::MetadataSections *registered = nullptr;
-
-void record(const swift::MetadataSections *sections) {
-  if (registered == nullptr) {
-    registered = sections;
-    sections->next = sections->prev = sections;
-  } else {
-    registered->prev->next = sections;
-    sections->next = registered;
-    sections->prev = registered->prev;
-    registered->prev = sections;
+  static const swift::MetadataSections *registered = nullptr;
+
+  void record(const swift::MetadataSections *sections) {
+    if (registered == nullptr) {
+      registered = sections;
+      sections->next = sections->prev = sections;
+    } else {
+      registered->prev->next = sections;
+      sections->next = registered;
+      sections->prev = registered->prev;
+      registered->prev = sections;
+    }
+  }
+
+  // Returns true if all bytes in the given memory range are 0.
+  bool areAllBytesZero(const char *start, std::size_t length) {
+    for (auto p = start; p < start + length; p++) {
+      if (*p != 0) {
+        return false;
+      }
+    }
+
+    return true;
+  }
```

Comment:
Please use `memcmp` instead of this.

Reply:
memcmp() compares two byte strings. Here we check whether all bytes in a memory region are 0. We would have to create a temporary byte string of all zeros with the size of 'recordStride' to be able to use memcmp(). I think that this approach here is simpler and clearer.